### ISSUE TITLE
Implement GammaCatDataCollection

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#2107]
https://gamma-cat.readthedocs.io/ is a dataset collection and source catalog for TeV astronomy. There is some code in Gammapy to make it easy to work with it: https://docs.gammapy.org/0.11/catalog/gammacat.html

I think the catalog and source class are pretty good already.

However, the `GammaCatResourceCollection` is completely unfinished, it doesn't implement the key feature to load a dataset the user wants via a method - such as "give me a gammapy.spectrum.FluxPoints object for this resource".

If anyone has time to implement this, that would be very nice. It's also a good issue to get started with Gammapy because the coding is only in one file ([gammapy/catalog/gammacat.py](https://github.com/gammapy/gammapy/blob/master/gammapy/catalog/gammacat.py)) and it's easy to test (add one case showing how it works in [gammapy/catalog/tests/test_gammacat.py](https://github.com/gammapy/gammapy/blob/master/gammapy/catalog/tests/test_gammacat.py), and document (edit [docs/catalog/gammacat.rst](https://github.com/gammapy/gammapy/blob/master/docs/catalog/gammacat.rst) or transform that RST file into a Jupyter tutorial notebook). This is also a good introduction if someone is interested to learn about open data formats (ECSV, JSON, YAML).

The work can proceed in steps - a first goal could be to make it easy to select and load a given spectrum. There is a very old (probably broken) notebook with example code that one could look at for guidance or copy & paste from here: https://github.com/cdeil/gamma-cat-status/blob/cc3084fc9b43933a93203275eed7385d322efb82/notebooks/data_collection_example.ipynb

**Comment by cdeil** [comment]
@registerrier @adonath - OK if I remove the gamma-cat collection?
https://docs.gammapy.org/0.14/catalog/gammacat.html#data-collection

(not the catalog and source class, just the collection)

With one full day of work I could finish it to a decent state, where all available methods work, and flux points & lightcurves can be loaded with a method call, and spectral and spatial model information is 
available as Python dicts ((see old [examples](https://github.com/cdeil/gamma-cat-status/blob/cc3084fc9b43933a93203275eed7385d322efb82/notebooks/data_collection_example.ipynb)). Mapping that information to Gammapy spectral and spatial objects would probably be another full day, maybe two.

For me, this is a close call whether to improve or remove, so looking for your feedback.

(otherwise, I'll send a PR to improve gammapy.catalog docs and tutorials shortly, leaving this alone for now)

**Comment by adonath** [comment]
@cdeil OK to remove it, I think not many users (if anyone at all) are using it...

**Comment by cdeil** [comment]
Removed in #2617. Closing issue.

> I think not many users (if anyone at all) are using it...

Well, that's because the implementation was unfinished and not documented.
People didn't know it existed or how to access the data.

There were requests (see https://github.com/gammapy/gamma-cat/issues/194) and I got several more via private email of people asking how they can select and load gamma-cat data.

So I don't think the statement that we should remove it because there's no user interest is fair.

But OK with the conclusion to remove it, just that my reason is I don't have time to finish it for v1.0, prefer to help and focus on other topics in Gammapy.
